src/config/config.cpp

```
@@ -540,3 +542,105 @@ void print_default_env() {
+std::vector<HostEntry> load_hosts_json(const std::string& path) {
+  std::vector<HostEntry> result;
+  std::ifstream f(path);
+  if (!f.is_open()) return result;
+  std::string content((std::istreambuf_iterator<char>(f)), std::istreambuf_iterator<char>());
+
+  // Simple array-of-objects parser using existing json helpers
+  size_t pos = 0;
+  while ((pos = content.find('{', pos)) != std::string::npos) {
+    std::string obj = json_extract_object_at(content, pos);
+    if (obj.empty()) break;
+    HostEntry e;
```

Clang (14.0.6)

[warning] variable name 'e' is too short, expected at least 3 characters

(readability-identifier-length)